src/catalua.cpp

```diff
@@ -124,16 +127,25 @@ bool lua_report_error(lua_State *L, int err, const char *path) {
     const std::string error = lua_tostring_wrapper( L, -1 );
     switch(err) {
         case LUA_ERRSYNTAX:
-            debugmsg( "Lua returned syntax error for %s\n%s", path, error.c_str() );
+            if(!simple) {
+                lua_error_stream << string_format( "Lua returned syntax error for %s", path ) << std::endl;
+            }
+            lua_error_stream << error;
             break;
         case LUA_ERRMEM:
-            debugmsg( "Lua is out of memory" );
+            lua_error_stream << string_format( "Lua is out of memory" );
```

Comment:
At least this string format call is completely useless, it returns the input as is without any changes.

And the call above with "%s" as path is a bit pointless as well, you can just forward the path and the c-string directly to the strings: `stream << "text " << path;`.

Same for the lines below.

Shouldn't there be new-lines or some of he separator at the of the string?